src/main/java/org/jenkinsci/plugins/workflow/steps/durable_task/DurableTaskStep.java

```diff
@@ -332,12 +321,10 @@ private void check() {
                             recurrencePeriod = 0;
                             controller.cleanup(workspace);
                         }
-                        return null;
-                    }
-                }, 10, TimeUnit.SECONDS);
             } catch (Exception x) {
                 LOGGER.log(Level.FINE, "could not check " + workspace, x);
                 ws = null;
+                logger().println("Cannot contact " + node + ": " + x); // TODO as above
```

Comment:
How to distinguish both cases?
What do you think about adding a configurable timeout to help detect that node is definitely offline (like done at connection I think).

for information, I created the next issue in jenkins bug tracker: https://issues.jenkins-ci.org/browse/JENKINS-43607